src/databricks/labs/ucx/source_code/notebook.py

```diff
@@ -0,0 +1,331 @@
+
+    @property
+    @abstractmethod
+    def language(self) -> CellLanguage:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def is_runnable(self) -> bool:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def build_dependency_graph(self, parent: NotebookDependencyGraph):
+        raise NotImplementedError()
+
+
+class PythonCell(Cell):
+
+    @property
+    def language(self):
+        return CellLanguage.PYTHON
+
+    def is_runnable(self) -> bool:
+        try:
+            ast = parse_python(self._original_code)
+            return ast is not None
+        except SyntaxError:
+            return False
+
+    def build_dependency_graph(self, parent: NotebookDependencyGraph):
+        pass  # TODO
+        # see https://github.com/databrickslabs/ucx/issues/1200
+        # see https://github.com/databrickslabs/ucx/issues/1202
+
+
+class RCell(Cell):
+
+    @property
+    def language(self):
+        return CellLanguage.R
+
+    def is_runnable(self) -> bool:
+        return True  # TODO
+
+    def build_dependency_graph(self, parent: NotebookDependencyGraph):
+        pass  # not in scope
+
+
+class ScalaCell(Cell):
+
+    @property
+    def language(self):
+        return CellLanguage.SCALA
+
+    def is_runnable(self) -> bool:
+        return True  # TODO
+
+    def build_dependency_graph(self, parent: NotebookDependencyGraph):
+        pass  # TODO
+
+
+class SQLCell(Cell):
+
+    @property
+    def language(self):
+        return CellLanguage.SQL
+
+    def is_runnable(self) -> bool:
+        try:
+            statements = parse_sql(self._original_code)
+            return len(statements) > 0
+        except SQLParseError:
+            return False
+
+    def build_dependency_graph(self, parent: NotebookDependencyGraph):
+        pass  # not in scope
+
+
+class MarkdownCell(Cell):
+
+    @property
+    def language(self):
+        return CellLanguage.MARKDOWN
+
+    def is_runnable(self) -> bool:
+        return True  # TODO
+
+    def build_dependency_graph(self, parent: NotebookDependencyGraph):
+        pass  # not in scope
+
+
+class RunCell(Cell):
+
+    @property
+    def language(self):
+        return CellLanguage.RUN
+
+    def is_runnable(self) -> bool:
+        return True  # TODO
+
+    def build_dependency_graph(self, parent: NotebookDependencyGraph):
+        command = f'{LANGUAGE_PREFIX}{self.language.magic_name}'.strip()
+        lines = self._original_code.split('\n')
+        for line in lines:
+            start = line.index(command)
+            if start >= 0:
+                path = line[start + len(command) :].strip()
+                parent.register_dependency(path.strip('"'))
+                return
+        raise ValueError("Missing notebook path in %run command")
+
+
+class CellLanguage(Enum):
+    # long magic_names must come first to avoid shorter ones being matched
+    PYTHON = Language.PYTHON, 'python', '#', PythonCell
```

Comment:
oh, we can do that?..

Reply:
you mean class enums ?
yes we can ! 
(I might win the next US elections with this catch phrase)

Reply:
didn't realise we can just pass a tuple in there